components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -105,14 +170,20 @@ data class NimbusServerSettings(
 /**
  * A implementation of the [NimbusApi] interface backed by the Nimbus SDK.
  */
+@Suppress("TooManyFunctions")
 class Nimbus(
-    context: Context,
+    private val context: Context,
     server: NimbusServerSettings?,
     private val delegate: Observable<NimbusApi.Observer> = ObserverRegistry()
 ) : NimbusApi, Observable<NimbusApi.Observer> by delegate {
 
-    // Using a single threaded executor here to enforce synchronization where needed.
-    private val scope: CoroutineScope =
+    // Using two single threaded executors here to enforce synchronization where needed:
```

Comment:
❤️ the changes here! I was about to ask about the two dedicated threads, because that's two more we create in Fenix and we already have 120+ :), but if you want to guarantee that you have a single dedicated thread for fetch and one for db operations that's the way...